sqlglot/dialects/tsql.py

```diff
@@ -526,6 +526,12 @@ class Parser(parser.Parser):
             *parser.Parser.TYPE_TOKENS,
         }
 
+        TYPE_CONVERTER = {
+            # TINYINT is a 1-byte unsigned integer
+            # https://learn.microsoft.com/en-us/sql/t-sql/data-types/int-bigint-smallint-and-tinyint-transact-sql?view=sql-server-ver16
+            exp.DataType.Type.TINYINT: lambda _: exp.DataType(this=exp.DataType.Type.UTINYINT),
```

Comment:
why do you need the type_converter, can't we use the type mapping?

Reply:
Which type mapping? Tokenizer?

Reply:
yea just map the tinyint token for tsql to uint

Reply:
done